Mathlib/NumberTheory/Cyclotomic/Basic.lean

```diff
@@ -542,12 +542,11 @@ instance CyclotomicField.algebra' {R : Type*} [CommRing R] [Algebra R K] :
 instance {R : Type*} [CommRing R] [Algebra R K] : IsScalarTower R K (CyclotomicField n K) :=
   SplittingField.isScalarTower _
 
-instance CyclotomicField.noZeroSMulDivisors [IsFractionRing A K] :
-    NoZeroSMulDivisors A (CyclotomicField n K) := by
-  refine NoZeroSMulDivisors.of_algebraMap_injective ?_
-  rw [IsScalarTower.algebraMap_eq A K (CyclotomicField n K)]
+instance [IsFractionRing A K] :
+    FaithfulSMul A (CyclotomicField n K) := by
```

Comment:
Was the slowdowns partially due to the need to synthesize NoZeroSMulDivisors from FaithfulSMul? Would it be better to keep the NoZeroSMulDivisors instances?

Reply:
I believe the slowdowns were attempts to go through `NoZeroSMulDivisors.instOfFaithfulSMul` when there was a more direct path or when it did not apply (e.g., when `NoZeroDivisors A` did not hold).

I have mostly tried not to refactor as I go but I made an exception here because its statement and proof really are about injectivity of `algebraMap`.

Reply:
On further reflection I have decided to reinstate the original form of this statement.